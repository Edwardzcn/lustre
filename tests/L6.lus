function divtest(x : int; y : int) returns (q : int; r : int)
let
  q = x div y;
  r = x mod y;
tel
