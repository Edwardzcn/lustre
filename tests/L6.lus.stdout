node (x_1 : int, y_2 : int)
  returns (q_3, r_4)
  assumes ()
  shows ()
let
  q_3 = Div(x_1, y_2)
  r_4 = Mod(x_1, y_2)
tel
--- Step 1 ---
x_1 : int  = 7
y_2 : int  = 3
q_3 = 2
r_4 = 1
--- Step 2 ---
x_1 : int  = 7
y_2 : int  = -3
q_3 = -2
r_4 = 1
--- Step 3 ---
x_1 : int  = -7
y_2 : int  = 3
q_3 = -3
r_4 = 2
--- Step 4 ---
x_1 : int  = -7
y_2 : int  = -3
q_3 = 3
r_4 = 2
